include/attributes_builder.h

```diff
@@ -0,0 +1,103 @@
+/* Copyright 2017 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MIXERCLIENT_ATTRIBUTES_BUILDER_H
```

Comment:
if we are still following Google style, this file is breaking it a lot, non-const reference parameters etc.
it might be better just use this as AttributesUtil make all methods static.

Reply:
Changed to use raw pointer for Builder constructor.
This Java style of .Add() has much cleaner code than static util functions.